# Market Analysis 

## 1. Import the required Libraries 

In [4]:
import pandas as pd 
import numpy as np 
import seaborn as sns 
import matplotlib.pyplot as plt 
%matplotlib inline 



## 2. Data Munging 

* Load the raw data to a readbale format

In [5]:
data = pd.read_csv(r"C:\Users\wanji\Desktop\edd(AutoRecovered).csv")
data

Company Name                Service Line  Service Line(%) Focus  \
0    BluChip Solutions LTD  Custom Software Development             40%         
1                      NaN       IT Strategy Consulting             20%         
2                      NaN              Web Development             20%         
3                      NaN       E-Commerce Development             10%         
4                      NaN       Mobile App Development             10%         
..                     ...                          ...             ...   ...   
154                    NaN                          NaN             NaN   NaN   
155   Litchman Consultants  Custom Software Development             40%   NaN   
156   Litchman Consultants          Business Consulting             20%   NaN   
157   Litchman Consultants       Mobile App Development             20%   NaN   
158   Litchman Consultants                   Web Design             20%   NaN   

    Focus(%)               Industries Industry(%)                 Clients  \
0        NaN                Education         20%  Small Business (<$10M)   
1        NaN                  Medical         15%  Midmarket ($10M - $1B)   
2        NaN               Non-Profit         15%       Enterprise (>$1B)   
3        NaN       Financial services         10%                     NaN   
4        NaN  Hospitality & leisure,          10%                     NaN   
..       ...                      ...         ...                     ...   
154      NaN                      NaN         NaN                     NaN   
155      NaN                      NaN         NaN                     NaN   
156      NaN                      NaN         NaN                     NaN   
157      NaN                      NaN         NaN                     NaN   
158      NaN                      NaN         NaN                     NaN   

    Clients(%)  
0          65%  
1          20%  
2          15%  
3          NaN  
4          NaN  
..         ...  
154        NaN  
155        NaN  
156        NaN  
157        NaN  
158        NaN  

[159 rows x 9 columns]

In [6]:
# First /Top entries of data 
data.head()

Company Name                Service Line  Service Line(%) Focus  \
0  BluChip Solutions LTD  Custom Software Development             40%         
1                    NaN       IT Strategy Consulting             20%         
2                    NaN              Web Development             20%         
3                    NaN       E-Commerce Development             10%         
4                    NaN       Mobile App Development             10%         

  Focus(%)               Industries Industry(%)                 Clients  \
0      NaN                Education         20%  Small Business (<$10M)   
1      NaN                  Medical         15%  Midmarket ($10M - $1B)   
2      NaN               Non-Profit         15%       Enterprise (>$1B)   
3      NaN       Financial services         10%                     NaN   
4      NaN  Hospitality & leisure,          10%                     NaN   

  Clients(%)  
0        65%  
1        20%  
2        15%  
3        NaN  
4        NaN

In [7]:
# Last entries of data 
data.tail()

Company Name                Service Line  Service Line(%) Focus  \
154                   NaN                          NaN             NaN   NaN   
155  Litchman Consultants  Custom Software Development             40%   NaN   
156  Litchman Consultants          Business Consulting             20%   NaN   
157  Litchman Consultants       Mobile App Development             20%   NaN   
158  Litchman Consultants                   Web Design             20%   NaN   

    Focus(%) Industries Industry(%) Clients Clients(%)  
154      NaN        NaN         NaN     NaN        NaN  
155      NaN        NaN         NaN     NaN        NaN  
156      NaN        NaN         NaN     NaN        NaN  
157      NaN        NaN         NaN     NaN        NaN  
158      NaN        NaN         NaN     NaN        NaN

##### 2.1 Dataset Info
What the dataset Generally Entails and in this case 
* Shape
* Columns , Column Names
* Data Type
* Statistics

In [8]:
# Know what the data entails
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 159 entries, 0 to 158
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Company Name     36 non-null     object
 1   Service Line     82 non-null     object
 2   Service Line(%)  89 non-null     object
 3   Focus            88 non-null     object
 4   Focus(%)         67 non-null     object
 5   Industries       62 non-null     object
 6   Industry(%)      61 non-null     object
 7   Clients          19 non-null     object
 8   Clients(%)       20 non-null     object
dtypes: object(9)
memory usage: 11.3+ KB


In [9]:
# Statistics of the dataset: 
data.describe()

Company Name                Service Line  Service Line(%) Focus  \
count              36                           82              89    88   
unique             14                           33              20    61   
top     DMI Analytics  Custom Software Development             10%         
freq               16                            6              31     8   

       Focus(%)  Industries Industry(%)                 Clients Clients(%)  
count        67          62          61                      19         20  
unique       11          30           9                       3         11  
top         10%  Non-profit         10%  Small Business (<$10M)        20%  
freq         21           7          27                       7          5

In [11]:
data.shape

(159, 9)

* From the above, the dataset contains 159 rows and 9 columns 

## Data Cleaning 
* Identify duplicates and drop them, but depending with the data 
* Identify missing values: Use count, mean or drop missing values depending with the data 

In [13]:
# Identifyy the duplicates 
data.duplicated().sum()

20

* From the above we have some duolicated values , but given the data there are duplicated elements but have different entries, hence no need for dropping them 